### PR TITLE
Add scripts to disable an app its services

### LINKED ISSUE #1349: Disable/remove apps
- Update service stats creator so that it ignores services of which expiredAt > 0 https://github.com/our-city-app/mobicage-backend/blob/b0bb0e0705c26f4d9a81c672b005d41881ec9b9c/src/rogerthat/bizz/job/service_stats.py#L52
- Send message to users that the app will disappear soon (app broadcast)
  - manually
- Gather emails from all merchants to message them
- Disable all services
- Add disabled property to 'App'
  - Omit disabled apps in the 'released apps' statistic
  - Don't allow registration when app is disabled
- remove from app store
  - manually
- remove from play store
  - manually

### PR DESCRIPTION
**Pull request by lucasvanhalst**
Closes #1349 

Needs https://github.com/our-city-app/mobicage-backend/pull/199

TODO: message to services
```
Beste,

Uw gemeente/stad X heeft beslist om niet langer in te zetten op de stadsapp. De samenwerking tussen gemeente/stad X en Onze Stad App werd jammer genoeg stopgezet.

Uw account zal verwijderd worden en het zal niet langer mogelijk zijn om in te loggen op uw dashboard. De app zal definitief verdwijnen op <datum>.

We willen u, handelaar, vereniging, zorgverstrekker, bedanken voor het gebruiken van de app, het vertrouwen en de prettige samenwerking.

Indien u nog vragen heeft hieromtrent kunt u ons steeds contacteren op het nummer 09/324 25 64 of op info@onzestadapp.be

Met vriendelijke groeten
Het Onze Stad App team
```

```python
from solutions.common.job.delete_app import _1_send_goodbye_message, _2_gather_emails, _3_set_app_disabled, _4_disable_all_customers

app_id = 'rogerthat'

def _1():
  message = """Beste,

Uw gemeente/stad %(app)s heeft beslist om niet langer in te zetten op de stadsapp. De samenwerking tussen gemeente/stad %(app)s en Onze Stad App werd jammer genoeg stopgezet.

De app zal definitief verdwijnen op <datum>. Vanaf dan is het niet langer mogelijk om de app te gebruiken. Bijgevolg kunt u de app verwijderen van uw telefoon of tablet.

We willen u bedanken voor het gebruiken van de app, het vertrouwen en de prettige samenwerking.

Indien u nog vragen heeft hieromtrent kunt u ons steeds contacteren op het nummer 09/324 25 64 of op info@onzestadapp.be

Met vriendelijke groeten
Het Onze Stad App team
"""
  return _1_send_goodbye_message(app_id, message)

def _2():
  return _2_gather_emails(app_id)

def _3():
  _3_set_app_disabled(app_id)
  
def _4():
  reason = 'tis ermee gedaan'
  return _4_disable_all_customers(app_id, reason, True)
```

### CODE REVIEW

File: src/solutions/common/job/delete_app.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from shop.bizz import post_app_broadcast, put_service
+from shop.business.order import cancel_subscription
+from shop.models import Customer
+from shop.view import _get_service
+from solution_server_settings import get_solution_server_settings
+
+
+def _1_send_goodbye_message(app_id, message):
+    app = get_app(app_id)
+    message = message % {'app': app.name}
```
Comment (RubenMattan) [inline src/solutions/common/job/delete_app.py]: the translation key is wrong, it should be 'app_name'

File: src/solutions/common/job/delete_app.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    post_app_broadcast(app.main_service, [app_id], message)
+
+
+def _2_gather_emails(app_id):
+    identities = ServiceIdentity.list_by_app_id(app_id)  # type: list[ServiceIdentity]
+    return [str(i.qualifiedIdentifier or i.identifier) for i in identities]
```
Comment (RubenMattan) [inline src/solutions/common/job/delete_app.py]: should be something like this [c.user_email for c in Customer.all().filter('app_ids =', app_id)]

File: src/solutions/common/job/delete_app.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    tasks = [create_task(cancel_subscription, c.id, cancel_reason=reason, immediately=True) for c in to_cancel]
+    schedule_tasks(tasks, MIGRATION_QUEUE)
+    tasks = [create_task(_remove_app_from_customer, c.id, app_id) for c in to_remove_app]
+    schedule_tasks(tasks, MIGRATION_QUEUE)
+
+
+def _remove_app_from_customer(customer_id, app_id):
+    admin = get_solution_server_settings().shop_bizz_admin_emails[0]
+    service = _get_service(customer_id, gusers.User(admin))
+    gusers.get_current_user()
```
Comment (RubenMattan) [inline src/solutions/common/job/delete_app.py]: remove

File: src/solutions/common/job/delete_app.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+def _2_gather_emails(app_id):
+    identities = ServiceIdentity.list_by_app_id(app_id)  # type: list[ServiceIdentity]
+    return [str(i.qualifiedIdentifier or i.identifier) for i in identities]
+
+
+def _3_set_app_disabled(app_id):
+    app = get_app(app_id)
+    app.disabled = True
+    app.put()
```
Comment (RubenMattan) [inline src/solutions/common/job/delete_app.py]: this should also call put_app_signup_enabled(app_id, False)

File: src/solutions/common/job/delete_app.py (hunk 3 above)
Comment (lucasvanhalst) [reply src/solutions/common/job/delete_app.py]: Not needed?

**Pull request merged**